include/snowflake/platform.h

```diff
@@ -144,6 +144,11 @@ void STDCALL sf_get_username(char * username, int bufLen);
 
 void STDCALL sf_delete_uniq_dir_if_exists(const char *tmpfile);
 
+void STDCALL sf_memory_error_handler();
+
+// this should be called by odbc driver before any calls of sfclient
```

Comment:
make the comment more general about what's the functionality or no comment as other functions. This function could be called in other case as well if the application prefer to get exception on memory allocation error.